router/projectRoute.js

```diff
@@ -4,61 +4,78 @@ const express = require('express');
 const projectData = require('../data/helpers/projectModel');
 const router = express.Router();
 
+
 // GET projects
 // test passes? yes
-router.get('/', (req, res) => {
+router.get('/', validateProjectParams, (req, res) => {
     projectData
         .get()
         .then(projects => res.status(200).json(projects))
         .catch(() => res.status(500).json({ message: 'The projects could not be retrieved from the database.'}));
 });
 
+
 //GET projects by id
 // test passes? yes, sort of
-router.get('/:id', (req, res) => {
+router.get('/:id', validateProjectParams, (req, res) => {
     const { id } = req.params
     projectData.get(id)
         .then(project =>  {
-            if(project.id > 0) res.status(200).json(project)
+            if(project.id) res.status(200).json(project)
             else res.status(400).json({ message: 'The project id could not be retrieved from the database.'}) 
-        .catch(() => res.status(404).json({ message: 'Projects could not be retrieved from the'}))
+        .catch(() => res.status(404).json({ message: 'Projects could not be retrieved from the database.'}))
```

Comment:
If you wanted to respond with the system error, as well as your own error message, you could structure your .catch this way:

```
 .catch(error => {
      res.status(500).json({
        message: "Projects could not be retrieved from the database.",
        error: `${error}`
      });
    });
```